Makefile

```diff
@@ -24,6 +24,31 @@ AUTHORS: .mailmap .git/HEAD
 PKGS_AND_MOCKS := $(shell go list ./... | grep -v /vendor)
 PKGS := $(shell echo $(PKGS_AND_MOCKS) | tr ' ' '\n' | grep -v /mock$)
 
+# Current working environment.  Set these explicitly if you want to cross-compile
+# in the build container (see the build-in-container target):
+
+GOOS?=$(shell go env GOOS)
+GOARCH?=$(shell go env GOARCH)
+VENDOR_HASH=$(shell git hash-object vendor/vendor.json)
+
+# First we create a container that is versioned (via the vendor.json hash) that has
+# all the downloaded vendor dependencies.  Then we use that container as the data
+# volume container across multiple builds, as long as the vendoring dependencies
+# don't chnage.  When vendor.json is updated, a new hash results in a new container
+# which is reused across builds again.  If you bulid this target, vendor/ directory
+# will not be changed.  Instead the vendored files will be in the container's fs namespace.
+build-in-container:
+	@echo "+ $@"
+	docker build -t infrakit-build:${VENDOR_HASH} -f ${CURDIR}/dockerfiles/Dockerfile.build .
+	-docker run --name infrakit-build-${VENDOR_HASH} \
+		-e GOOS=${GOOS} -e GOARCCH=${GOARCH} \
+		infrakit-build:${VENDOR_HASH}
```

Comment:
^ oops, still getting used to GitHub reviews, meant to post this in a separate "conversation"